src/fusiontensor/base_interface.jl

```diff
@@ -0,0 +1,165 @@
+# This files defines Base functions for FusionTensor
+
+using BlockSparseArrays: @view!
+
+function Base.:*(x::Number, ft::FusionTensor)
+  return FusionTensor(
+    x * data_matrix(ft), codomain_axes(ft), domain_axes(ft), trees_block_mapping(ft)
+  )
+end
+
+function Base.:*(ft::FusionTensor, x::Number)
+  return FusionTensor(
+    x * data_matrix(ft), codomain_axes(ft), domain_axes(ft), trees_block_mapping(ft)
+  )
+end
+
+# tensor contraction is a block data_matrix product.
+function Base.:*(left::FusionTensor, right::FusionTensor)
+  @assert matching_dual(domain_axes(left), codomain_axes(right))
+  new_data_matrix = data_matrix(left) * data_matrix(right)
+  return fusiontensor(new_data_matrix, codomain_axes(left), domain_axes(right))
+end
+
+Base.:+(ft::FusionTensor) = ft
+
+# tensor addition is a block data_matrix add.
+function Base.:+(left::FusionTensor, right::FusionTensor)
+  @assert matching_axes(axes(left), axes(right))
```

Comment:
Probably would be good to wrap this into a function, and throw a more specific error:
```julia
function check_matching_axes(ax1, ax2)
  matching_axes(ax1, ax2) || throw(DimensionMismatch("axes don't match."))
  return nothing
end
```

Reply:
If you want to go the `Base` way, similar to `checkbounds`, you could define
```julia
checkaxes(Bool, ax1, ax2) -> Bool
checkaxes(ax1, ax2) = checkaxes(Bool, ax1, ax2) || throw(DimensionMismatch(lazy"$ax1 does not match $ax2
))
```